src/Core/Checkout/Cart/Delivery/Struct/DeliveryCollection.php

```diff
@@ -82,6 +84,13 @@ public function getAddresses(): CustomerAddressCollection
         return $addresses;
     }
 
+    public function getPrimaryDelivery(?string $primaryDeliveryId): ?Delivery
+    {
+        return $this->firstWhere(function (Delivery $delivery) use ($primaryDeliveryId) {
```

Comment:
please add the check for returning the `first` here and also add a comment above, explaining the behaviour